XcodeAutoBasher/XcodeAutoBasher/Models/VOKProjectContainer.m

```diff
@@ -116,6 +116,52 @@ - (VOKDirectoryWatcher *)directoryWatcher
     return _directoryWatcher;
 }
 
+- (NSDictionary *)environmentVariables
+{
+    if ([NSThread isMainThread]) {
+        NSLog(@"VOKProjectContainer environmentVariables being retrieved on the main thread.");
+    }
+    NSPipe *outputPipe = [NSPipe pipe];
+    
+    NSTask *buildSettingsTask = [[NSTask alloc] init];
+    buildSettingsTask.launchPath = @"/usr/bin/xcodebuild";
+    buildSettingsTask.currentDirectoryPath = self.containingPath;
+    buildSettingsTask.arguments = @[ @"-showBuildSettings", ];
+    buildSettingsTask.standardOutput = outputPipe;
+    
+    [buildSettingsTask launch];
+    [buildSettingsTask waitUntilExit];
+    
+    NSData *outputData = [outputPipe.fileHandleForReading readDataToEndOfFile];
+    NSString *outputString = [[NSString alloc] initWithData:outputData encoding:NSUTF8StringEncoding];
+    
+    NSError *error;
+    NSRegularExpression *regex = [NSRegularExpression
```

Comment:
They are only voodoo when using lookarounds and stuff like that.

I'd say this is loosely based on that regex, at best, and follows from seeing that the output of `xcodebuild -showBuildSettings` is a list of lines of the form `[whitespace][key][whitespace]=[whitespace][value]`, but I'll put in whichever/whatever documenting comment you want here.